pkg/remediator/conflict/handler.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conflict
+
+import (
+	"sync"
+
+	"k8s.io/klog/v2"
+	"kpt.dev/configsync/pkg/status"
+)
+
+// Handler is the generic interface of the conflict handler.
+type Handler interface {
+	AddConflictError(status.ManagementConflictError)
+	RemoveConflictError(status.ManagementConflictError)
+
+	// ConflictErrors returns the management conflict errors (KNV1060) the remediator encounters.
+	ConflictErrors() []status.ManagementConflictError
+}
+
+// handler implements Handler.
+type handler struct {
+	// mux guards the conflictErrs
+	mux sync.Mutex
+	// conflictErrs tracks all the conflict errors (KNV1060) the remediator encounters,
+	// and report to RootSync|RepoSync status.
+	conflictErrs []status.ManagementConflictError
+}
+
+var _ Handler = &handler{}
+
+// NewHandler instantiates a conflict handler
+func NewHandler() Handler {
+	return &handler{
+		conflictErrs: []status.ManagementConflictError{},
+	}
+}
+
+func (h *handler) AddConflictError(e status.ManagementConflictError) {
+	h.mux.Lock()
+	defer h.mux.Unlock()
+
+	for _, existingErr := range h.conflictErrs {
+		if e.Error() == existingErr.Error() {
+			return
+		}
+	}
+	h.conflictErrs = append(h.conflictErrs, e)
```

Comment:
Should we sort on add so they stay in a consistent order? Or are they always just ordered by order of appearance?

Reply:
What about sorting on get?

Reply:
Switched to use orderedmap.